scipy/linalg/_decomp_svd.py

```diff
@@ -133,22 +133,35 @@ def svd(a, full_matrices=True, compute_uv=True, overwrite_a=False,
     if lapack_driver not in ('gesdd', 'gesvd'):
         message = f'lapack_driver must be "gesdd" or "gesvd", not "{lapack_driver}"'
         raise ValueError(message)
-
-    if lapack_driver == 'gesdd' and compute_uv:
+    funcs = (lapack_driver, lapack_driver + '_lwork')
+    try:
+        gesXd, gesXd_lwork = get_lapack_funcs(funcs, (a1,), ilp64=True)
+        ilp64_enabled = True
+    except RuntimeError:
+        # some debug warning here
+        gesXd, gesXd_lwork = get_lapack_funcs(funcs, (a1,), ilp64=False)
+        ilp64_enabled = False
+
+    if compute_uv and not ilp64_enabled:
         # XXX: revisit int32 when ILP64 lapack becomes a thing
         max_mn, min_mn = (m, n) if m > n else (n, m)
         if full_matrices:
             if max_mn*max_mn > np.iinfo(np.int32).max:
                 raise ValueError(f"Indexing a matrix size {max_mn} x {max_mn} "
-                                  " would incur integer overflow in LAPACK.")
+                                  "would incur integer overflow in LAPACK. "
+                                  "A possible workaround is to 1. build NumPy "
+                                  "against ILP64 LAPACK (see https://numpy.org"
+                                  "/doc/stable/building/blas_lapack.html) "
```

Comment:
I'm pretty sure official numpy binaries use ilp64. Regardless, scipy should not suggest to users to rebuild numpy from source :-). Better soften the wording to just "try using numpy svd" or keep the warning intact.

Reply:
I think you are right. I tried
```
python -c "import numpy as np; a = np.ones((10_000_000, 256), dtype=np.float32); np.linalg.svd(a, full_matrices=False)"
```
in a clean Python 3.11 environment (just `pip install numpy`) and it _didn't_ crash

Reply:
I changed the message so simply suggest numpy.linalg.svd